File: COOLSOUND-BACKEND/coolsound/src/main/java/com/java/coolsound/service/ComentarioCancionService.java
package com.java.coolsound.service;

import java.util.List;

import com.java.coolsound.model.ComentarioCancion;

public interface ComentarioCancionService {

	/**
	 * Gets the all ComentarioCancion.
	 *
	 * @return the all ComentarioCancion
	 */
	List<ComentarioCancion> getAllComentarioCancion(int idCancion);
	
	/**
	 * Eliminar ComentarioCancion.
	 *
	 * @return eliminar ComentarioCancion
	 */
	int eliminarComentarioCancion(int idComentarioCancion);
	
	/**
	 * Añadir ComentarioCancion.
	 *
	 * @return añadir ComentarioCancion
	 */
	int nuevoComentarioCancion(String textoComentario, int idUsuario, int idCancion);
	
}

File: COOLSOUND-FRONTEND/coolsound/src/app/services/comentariocancion.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Comentario } from '../model/comentario';

@Injectable({
  providedIn: 'root'
})
export class ComentariocancionService {

  private baseUrl = '/coolsound/comentarioCancion/';

  constructor(private http: HttpClient) { }

  /** Método para sacar los Comentarios de BD */
  comentario(id: number): Observable<any> {
    return this.http.get<Comentario>(`${this.baseUrl}` + 'getComentarioCancion/' + id);
  }

  /** Método para eliminar Comentarios de BD */
  eliminarComentario(idComentarioCancion: number): Observable<any> {
    return this.http.get(`${this.baseUrl}` + 'eliminarComentarioCancion/' + idComentarioCancion);
  }

  /** Método para añadir Comentarios a BD */
  nuevoComentario(texto: string, idUsuario: number, idCancion: number): Observable<any> {
    return this.http.get(`${this.baseUrl}` + 'nuevoComentarioCancion/' + texto + '/' + idUsuario + '/' + idCancion);
  }
}
